### ISSUE TITLE
[svg-native] [13. Gradients and Patterns] gradientUnits in SVG Native

### ISSUE THREAD

**Issue by zauguin** [w3c/svgwg#746]
According to the current [SVG Native editor's draft](https://svgwg.org/specs/svg-native/#pservers):

> The following attributes are forbidden:
> 
>     gradientUnits
> 
>     'xlink:href' on linearGradient or radialGradient
> 
> Delete section 13.3: Patterns.

Given that `gradientUnits` has default value `objectBoundingBox`, this seems to imply that a gradient always uses `objectBoundingBox` in SVG Native. But multiple comments here seem to imply something else, that `objectBoundingBox` is not supported and instead only `userSpaceOnUse` should be used with SVG Native. This also makes more sense: `userSpaceOnUse` is much easier to implement than `objectBoundingBox`, so I think it should be allowed. 

What is the current position on this? Are gradients with (implicit) `gradientUnits=objectBoundingBox` a required feature of SVG Native?

**Comment by msand** [comment]
I think objectBoundingBox only requires one extra multiplication by the bounds, if the values are unitless numbers no? Or is there more complexity to it?

**Comment by dirkschulze** [comment]
`objectBoundingBox` requires to know the bounds of a shape and put gradient data in perspective to it.

Computing the bounding box is the none-trivial part here and the reason why the current proposal of SVG Native does not require `objectBoundingBox`.

**Comment by zauguin** [comment]
@dirkschulze Ok. How should this work in practise? Does SVG Native change the default to `userSpaceOnUse`? This would be a bit surprising because other parts always seem to be compatible to SVG 1.1, but the alternative would be to explicitly specify the forbidden attribute `gradientUnits="userSpaceOnUse"`.

**Comment by AmeliaBR** [comment]
Yes, we definitely don't want to change the default value if the attribute isn't present. I'd missed that the current draft implies that behavior & I don't think we've discussed it in the group. Every other decision about SVG Native has been based on the idea that a valid SVG Native document would render identically as a regular SVG.

Dirk proposed a compromise [here](https://github.com/w3c/svgwg/issues/657#issuecomment-482778343): make `gradientUnits="userSpaceOnUse"` required in the markup; if gradients are used and the attribute isn't present, the file is no longer valid SVG Native. Rendering tools would not have to implement the `objectBoundingBox` behavior.

I don't love that solution, but it's better than changing the default, if there are strong implementer objections to bounding-box gradients.

**Comment by msand** [comment]
Requiring gradientUnits="userSpaceOnUse" seems sensible given the requirement of rendering identically in regular svg viewers. Especially if nothing else in the profile requires bounding box calculations at this point. 

Content with objectBoundingBox can be converted by precomputing the bounds and resolving the relevant attributes to userSpaceOnUse units instead. 

Seems like a reasonable simplification, not lovely to make gradients more verbose, but the cost of specifying/parsing the unit seems relatively small in comparison to implementing bounding box calculations and computing it at runtime for sure.

**Comment by dirkschulze** [comment]
In our last discussion (will update with minutes when I find them) we rather require setting `gradientUnits="userSpaceOnUse"` for a valid SVG Native document than "falling back" to `userSpaceOnUse `.

Note that the spec currently is in the progress of updating to the last resolutions. So not everything might be adapted in the spec text yet.
